bolt-sidecar/src/driver.rs

```diff
@@ -245,13 +273,18 @@ impl<C: StateFetcher, ECDSA: SignerECDSA> SidecarDriver<C, ECDSA> {
             "Validation against execution state passed"
         );
 
-        // parse the request into constraints and sign them
-        let slot = inclusion_request.slot;
-
-        let pubkey = match self.constraint_signer {
-            SignerBLS::Local(ref signer) => signer.pubkey(),
-            SignerBLS::CommitBoost(ref signer) => signer.pubkey(),
-            SignerBLS::Keystore(_) => validator_pubkey.clone(),
+        let delegatees = self.constraints_client.find_delegatees(&validator_pubkey);
+        let available_pubkeys = self.constraint_signer.available_pubkeys();
+
+        // Pick a pubkey to sign constraints with.
+        //
+        // Rationale:
+        // - If there are no delegatee keys, try to use the validator key directly if available.
+        // - If there are delegatee keys, try to use the first one that is available in the list.
```

Comment:
The `pick_public_key` function is already documented, if you want you can drop this!